js/gdm/authPrompt.js

```diff
@@ -592,8 +609,15 @@ var AuthPrompt = class {
         // version had one less digit in the code).
         let resetCode = Main.customerSupport.passwordResetSalt ? '1' : '';
 
-        for (let n = 0; n < _RESET_CODE_LENGTH; n++)
-            resetCode = '%s%d'.format(resetCode, GLib.random_int_range(0, 10));
+        let machineId = _getMachineId();
+        let lastLoginTime = this._getUserLastLoginTime();
+        let input = machineId + this._username + lastLoginTime;
+        let checksum = GLib.compute_checksum_for_data(GLib.ChecksumType.SHA256, input);
+        checksum = checksum.replace(/\D/g, '');
+
+        let hashCode = `${parseInt(checksum) % (10 ** _RESET_CODE_LENGTH)}`;
```

Comment:
At some point in the past it was the case that backtick strings would break `xgettext` and prevent it from seeing any translatable strings later in the file. You can run `ninja meson-gnome-shell-pot` and check that the translatable strings below appear in the `po/gnome-shell.pot` file to verify whether this is still the case.

Reply:
Checked and it seems to work fine. Kept as is.